ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResourceTest.java

```diff
@@ -257,51 +263,64 @@ public void shouldReturnServiceUnavailableIfTimeoutWaitingForCommandSequenceNumb
   }
 
   @Test
-  public void shouldNotCreateAdminClientForPullQuery() throws Exception {
-    // When:
-    testResource.streamQuery(
-        serviceContext,
-        new KsqlRequest(PULL_QUERY_STRING, Collections.emptyMap(), null)
-    );
+  public void shouldNotCreateExternalClientsForPullQuery() {
+    // Given
+    testResource.configure(new KsqlConfig(ImmutableMap.of(
+        StreamsConfig.APPLICATION_SERVER_CONFIG, "something:1",
+        KsqlConfig.KSQL_PULL_QUERIES_SKIP_ACCESS_VALIDATOR_CONFIG, true
+    )));
 
-    // Then:
+    // Expect
     verify(serviceContext, never()).getAdminClient();
-  }
+    verify(serviceContext, never()).getConnectClient();
+    verify(serviceContext, never()).getSchemaRegistryClient();
+    verify(serviceContext, never()).getTopicClient();
 
-  @Test
-  public void shouldNotCreateConnectClientForPullQuery() throws Exception {
     // When:
     testResource.streamQuery(
         serviceContext,
         new KsqlRequest(PULL_QUERY_STRING, Collections.emptyMap(), null)
     );
-
-    // Then:
-    verify(serviceContext, never()).getConnectClient();
   }
 
   @Test
-  public void shouldNotCreateSRClientForPullQuery() throws Exception {
+  public void shouldThrowExceptionForPullQueryIfValidating() {
     // When:
-    testResource.streamQuery(
+    final Response response = testResource.streamQuery(
         serviceContext,
         new KsqlRequest(PULL_QUERY_STRING, Collections.emptyMap(), null)
     );
 
     // Then:
-    verify(serviceContext, never()).getSchemaRegistryClient();
+    assertThat(response.getStatus(), is(Errors.badRequest("").getStatus()));
+    final KsqlErrorMessage expectedEntity = (KsqlErrorMessage) response.getEntity();
```

Comment:
consider adding `assertThat(response.getEntity(), is(instanceOf(KsqlErrorMessage.class)));` before this line to get a better error message than a class cast exception

Reply:
done.